homeassistant/helpers/trigger_template_entity.py

```diff
@@ -176,35 +199,81 @@ def restore_attributes(self, last_state: State) -> None:
                 extra_state_attributes[attr] = last_state.attributes[attr]
             self._rendered[CONF_ATTRIBUTES] = extra_state_attributes
 
-    def _render_templates(self, variables: dict[str, Any]) -> None:
-        """Render templates."""
+    def _render_single_template(
+        self,
+        key: str,
+        variables: dict[str, Any],
+        strict: bool = False,
+    ) -> Any:
+        """Render a single template."""
         try:
-            rendered = dict(self._static_rendered)
+            if key in self._to_render_complex:
+                return render_complex(self._config[key], variables)
 
-            for key in self._to_render_simple:
-                rendered[key] = self._config[key].async_render(
-                    variables,
-                    parse_result=key in self._parse_result,
-                )
+            return self._config[key].async_render(
+                variables, parse_result=key in self._parse_result, strict=strict
+            )
+        except TemplateError as err:
+            log_triggered_template_error(self.entity_id, err, key=key)
 
-            for key in self._to_render_complex:
-                rendered[key] = render_complex(
-                    self._config[key],
-                    variables,
-                )
+        return _SENTINEL
 
-            if CONF_ATTRIBUTES in self._config:
-                rendered[CONF_ATTRIBUTES] = render_complex(
-                    self._config[CONF_ATTRIBUTES],
-                    variables,
+    def _render_templates(self, variables: dict[str, Any]) -> None:
+        """Render templates."""
+        rendered = dict(self._static_rendered)
+
+        # Check availability first and render as a simple template because
+        # availability should only be able to render True, False, or None.
+        available = True
+        if CONF_AVAILABILITY in self._to_render_simple:
+            if (
+                result := self._render_single_template(
+                    CONF_AVAILABILITY, variables, strict=True
```

Comment:
Why is the strict flag set here and below when rendering complex templates?

Reply:
It's strict because it will resolve false if there's a template error.  We need to catch error's so that the template entity doesn't go unavailable when there's an exception in availability.  Trigger based sensors has this as a requirement.  Apparently in the past there was a PR that purposely made this functionality behave this way for regular template entities and now it's a requirement for trigger based as well.

Reply:
OK, but what about the complex templates?